helper/keysutil/policy.go

```diff
@@ -739,7 +813,29 @@ func (p *Policy) VerifySignature(hashedInput []byte, sig string) (bool, error) {
 			Y:     keyParams.EC_Y,
 		}
 
-		return ecdsa.Verify(key, hashedInput, ecdsaSig.R, ecdsaSig.S), nil
+		return ecdsa.Verify(key, input, ecdsaSig.R, ecdsaSig.S), nil
+
+	case KeyType_ED25519:
+		sigBytes, err := base64.StdEncoding.DecodeString(splitVerSig[1])
+		if err != nil {
+			return false, errutil.UserError{Err: "invalid base64 signature value"}
+		}
+
+		var key ed25519.PrivateKey
```

Comment:
Might be a bit of stretch in this case since it's just a couple of lines of code, but maybe could also dedup in here?